internal/mode/static/state/dataplane/configuration_test.go

```diff
@@ -2289,12 +2312,90 @@ func TestBuildConfiguration(t *testing.T) {
 					expHRWithPolicyGroups[0],
 					expHTTPSHRWithPolicyGroups[0],
 				},
-				BaseHTTPConfig: BaseHTTPConfig{
-					HTTP2: true,
-				},
+				BaseHTTPConfig: BaseHTTPConfig{HTTP2: true, IPFamily: Dual},
 			},
 			msg: "Simple Gateway and HTTPRoute with policies attached",
 		},
+		{
+			graph: &graph.Graph{
```

Comment:
There's starting to be a decent amount of duplication in this file. It would be nice if we could accept a boilerplate Graph and ExpectedConf that we could then use a `mod()` function to make the changes needed per test case, like we use in other tests.

I would say this could be a tech-debt story, but we work on this file so much I feel like we should just do it.

Reply:
Can you do this refactor as part of this? I don't expect it to be too much, just de-duplicating the boilerplate configs that are used and adding a `mod()` function for the specific changes needed per test.

Reply:
Yeah, i've done the refactoring as part of this code.